src/rust/plugin-registry/src/deploy_plugin.rs

```diff
@@ -0,0 +1,44 @@
+use std::collections::HashMap;
+
+use crate::{
+    nomad_cli,
+    nomad_client,
+    static_files,
+};
+
+/// https://github.com/grapl-security/grapl-rfcs/blob/main/text/0000-plugins.md#deployplugin-details
+#[tracing::instrument(skip(client, plugin_id), err)]
+pub async fn deploy_plugin(
+    client: nomad_client::NomadClient,
+    plugin_id: uuid::Uuid,
+) -> Result<(), nomad_client::NomadClientError> {
```

Comment:
I couldn't quite figure out how to structure the Errors in this whole crate.

deploy_plugin can have errors from Nomad Client but also from Nomad CLI.

Should it just be, everything in this service gets normalized to a PluginRegistryServiceError?

Reply:
Yeah, generally if you have errors from two different modules you create a third error that encapsulates them.

Reply:
I think the most important thing is to make sure we don't surface "internal" errors (e.g. nomad CLI/client error text) to consumers of the RPC method. So at the RPC layer we should translate internal errors to something appropriate for public consumption.

Reply:
Yep, that already happens